### ISSUE TITLE
Small Workgroup sizes negatively affecting performance for Batch Norm Kernels

### ISSUE THREAD

**Issue by YashDeshpande25** [iree-org/iree#23655]
This issue is a follow-up triage of the previous issues [#21824](https://github.com/iree-org/iree/issues/21824#issuecomment-3955467083) & related to [#23539](https://github.com/iree-org/iree/issues/23539).

Batch Norm produces a total of 5 dispatches, 2 of which perform reduction. On these dispatches, the workgroup size set by the heuristic for different Batch-Norm shapes when using TileAndFuse pipeline is equal to the size of the subgroup. 
Performance improves when using a larger workgroup size with smaller thread tile. 

### Original Configuration
```
{translation_info = ...workgroup_size = [64, 1, 1] subgroup_size = 64>}
...
llinalg.generic ... attrs = #iree_gpu.lowering_config<{reduction = [0, 4], thread = [4, 0], workgroup = [256, 0]}>}
```
mean execution time = 176.26us
[Link](https://gist.github.com/YashDeshpande25/8ce64bdc914fdc3ca1a90106e95d7ed1.js) to the complete unmodified IR 

### Manually Edited Configuration
```
{translation_info = ...workgroup_size = [256, 1, 1] subgroup_size = 64>}
...
llinalg.generic ... attrs = #iree_gpu.lowering_config<{reduction = [0, 4], thread = [1, 0], workgroup = [256, 0]}>}
```
mean execution time = 102.67us
[Link](https://gist.github.com/YashDeshpande25/6ac75637618378506e62377154a76a59) to the manually updated IR 

Timed using the boo-driver. 

## Steps to reproduce 

Running batchnorm config 12 via the boo-driver:
```
iree-boo-driver bnormbfp16 --batchsize 16 --in_channels 2048 --in_h 48 --in_w 32 --time 1 --iter 100 --save 1 --run 1 --layout NHWC
```

To compile the generated file 

- Use the flag `--compile-to=executable-configurations` to generate the editable IR. 

After manually modifying the IR, to generate an updated vmfb file from the updated mlir file

- Use the flag `--compile-from=executable-configurations` 

Use the flag `--iree-codegen-llvmgpu-use-reduction-vector-distribution=false` to apply the TileAndFuse pipeline
